cpp/src/gandiva/configuration.h

```diff
@@ -65,6 +71,9 @@ class GANDIVA_EXPORT Configuration {
   bool target_host_cpu_; /* set the mcpu flag to host cpu while compiling llvm ir */
   std::shared_ptr<FunctionRegistry>
       function_registry_; /* function registry that may contain external functions */
+  // flag indicating if IR dumping is needed, defaults to false, and turning it on will
+  // negatively affect performance
+  bool needs_ir_dumping_ = false;
```

Comment:
Do we need `needs_` prefix? It seems that we can just use `dump_ir` or something.

Reply:
Not problem. I don't like the prefix but I am not sure if `dump_ir` is too simple. Renamed them all to use `dump_ir`